app/models/flexible_criterion.rb

```diff
@@ -163,4 +158,42 @@ def visible?
     true
   end
 
+  def set_mark_by_criteria(mark_to_change, criterion_name)
+    if criterion_name == 'nil'
+      mark_to_change.mark = nil
+    else
+      mark_to_change.mark = criterion_name.to_f
+    end
+    mark_to_change.save
+  end
+
+  # Calculates the maximum mark that could be achieved considering all criteria marked
+  def total_mark
+    self.assignment.get_criteria(:ta).map(&:max).sum().round(2)
```

Comment:
Again, this calculation doesn't belong here. Do the sum at the point where you already have access to all the criteria.